### ISSUE TITLE
False warning for highways connected only to unloaded highways

### ISSUE THREAD

**Issue by SilentSpike** [openstreetmap/iD#5938]
See this scenario:
![image](https://user-images.githubusercontent.com/5459452/53269609-3ec2b500-36e1-11e9-8dd7-9409a8d7ec4b.png)

If you open location `map=18/57.08568/-2.14301` and hover over the highway, it is flagged as disconnected. In reality, it is connected, but just so long that (I assume) iD hasn't loaded the data where it connects to the rest of the network.

**Comment by quincylvania** [comment]
@SilentSpike Yep, I've seen this plenty of times in testing. It'd be nice to resolve this accurately by downloading more nearby/connecting data (see #5180). In the meantime we could ignore disconnected highways that overlap non-downloaded tiles.

**Comment by bhousel** [comment]
related #2248

**Comment by pnorman** [comment]
> @SilentSpike Yep, I've seen this plenty of times in testing. It'd be nice to resolve this accurately by downloading more nearby/connecting data (see #5180). 

You'd have to issue map calls along the length of the way, which could get expensive. You'll also then up with a bunch of new ways which exit the downloaded area.

> In the meantime we could ignore disconnected highways that overlap non-downloaded tiles.

I think this is the better choice. Errors of this type outside the downloaded area were obviously not created in the current editing session.

**Comment by bhousel** [comment]
> In the meantime we could ignore disconnected highways that overlap non-downloaded tiles.

I just added this to #6140  🎉 

I also noticed that both the `disconnected_ways` and `almost_junction` validations are vulnerable to this, so I'm fixing it in both places. 

In this example below, the left tile was loaded first, and an `almost_junction` warning was raised before the right tile loaded.

<img width="997" alt="Screenshot 2019-04-10 12 14 02" src="https://user-images.githubusercontent.com/38784/55899788-ababe480-5b93-11e9-8267-0ed5eed7c371.png">

I'll think more about this too, because anyplace in the code that we test the `parentWays` of a vertex that might not be on the visible map is risky.  

For example, we check `parentWays` of vertices in the turn restriction editor, but the vertex has to be visible and selected for iD to even do anything.  Might be worth adding defensive code anyway, in case the intersection extends off the screen and into unloaded map.
